Library/Homebrew/api.rb

```diff
@@ -69,8 +71,9 @@ def self.fetch_json_api_file(endpoint, target: HOMEBREW_CACHE_API/endpoint,
           args = curl_args.dup
           args.prepend("--time-cond", target.to_s) if target.exist? && !target.empty?
           if insecure_download
-            opoo "Using --insecure with curl to download #{endpoint} " \
-                 "because we need it to run `brew install ca-certificates`. " \
+            package = DevelopmentTools.curl_handles_most_https_certificates? ? "ca-certificates" : "curl"
```

Comment:
This might be a nice new function in `DevelopmentTools` given we use the same logic in two places. Could even handle printing the `opoo` too.